src/Inventory.java

```diff
@@ -0,0 +1,275 @@
+                "1 : Print current inventory\n" +
+                "2 : Create a new inventory item\n" +
+                "3 : Delete an item from inventory\n" +
+                "4 : Sell an item from inventory\n" +
+                "5 : Put item on sale\n" +
+                "6 : Change the price of an item\n" +
+                "7 : Change the quantity in stock of an item\n" +
+                "8 : Exit the program";
+
+        System.out.println(str);
+
+        int input = getValidInt(this.input.nextLine());
+
+        while (input < 1 || input > 8){
+            System.out.println("Please enter a valid option.");
+            input = getValidInt(this.input.nextLine());
+        }
+
+        switch(input) {
+            case 1:
+                printAll();
+                break;
+            case 2:
+                newItem();
+                break;
+            case 3:
+                deleteItem();
+                break;
+            case 4:
+                sellItem();
+                break;
+            case 5:
+                itemOnSale();
+                break;
+            case 6:
+                updatePrice();
+                break;
+            case 7:
+                updateQuantity();
+                break;
+            case 8:
+                exit();
+                break;
+        }
+
+
+    }
+
+    private void newItem() {
+        
+        System.out.println("\nWhat is the name of the item?");
+        String name = getValidStr(this.input.nextLine());
+        
+        System.out.println("\nWhat is the quantity?");
+        int qty = getValidQty(this.input.nextLine());
+        
+        System.out.println("\nWhat is the price?");
+        double price = getValidPrice(this.input.nextLine());
+        
+        Item item  = new Item(name, qty, price);
+        this.inventory.put(CURRENT_ITEM_ID, item);
+        CURRENT_ITEM_ID++;
+
+        System.out.format("\n%d <%s> @ %.2f has been added to inventory.\n", item.getQuantity(), item.getName(), item.getPrice());
+
+        whatYouWant();
+    }
+
+    private void deleteItem(){
+
+        int id =0;
+        while (id == 0) {
+            System.out.println("\nWhat is the ID of the item you would like to delete?");
+            id = getValidInt(this.input.nextLine());
+            if (this.inventory.get(id) == null){
+                System.out.println("\nNo such item exists at that ID.\n");
+                id = 0;
+            }
+        }
+
+        this.inventory.remove(id);
+        whatYouWant();
+    }
+
+    private void sellItem(){
+        Item item = getItem();
+        int itemQty = item.getQuantity();
+        System.out.println("\nWhat is the quantity to sell?");
+        int qty = getValidQty(this.input.nextLine());
+
+        while (qty > itemQty) {
+            System.out.println("\nThere is not enough in stock!");
+            System.out.println("\nWhat is the quantity to sell?");
+            qty = getValidQty(this.input.nextLine());
```

Comment:
This should be a red flag for you in that you have a few lines that are the same so this method could be refactored to be a bigger while loop to avoid the code duplication